### ISSUE TITLE
about tooltipHtml

### ISSUE THREAD

**Issue by sanyuelanv** [codesuki/react-d3-components#95]
i don't know why my chrome say that 
```
xScale.invert is not a function
```
in react 15

**Comment by ryanhca** [comment]
+1 This is happening for me as well. I can provide code samples of what's going on for me if needed. If you don't mind me asking @sanyuelanv... I'm using the linegraph for the tooltip functionality. Are you using the same?

**Comment by codesuki** [comment]
Thanks for the feedback! Could you provide a code sample of that happening. I'll have a look!

**Comment by ryanhca** [comment]
https://gist.github.com/ryanhca/6b8c1f9601abc2626696d2f1668ce362
<img width="1277" alt="screenshot 2016-07-03 09 50 52" src="https://cloud.githubusercontent.com/assets/8798259/16546314/f4c8e3a0-4103-11e6-84a6-659d2a83a703.png">

**Comment by codesuki** [comment]
Thanks I will have a look at it!

**Comment by codesuki** [comment]
Could I have a sample of your data? 
Doesn't need to be the original data, just the same format as the input you provide.

**Comment by ryanhca** [comment]
Sure. I build an array of these exact values. This is actually the exact format my data gets set to. The day's are simple buckets of time stamps formatted to Unix time.

totalLockData.values.push(
						{'x':moment().subtract(6, 'days').format('ddd MMMM Do'),'y':day7},
						{'x':moment().subtract(5, 'days').format('ddd MMMM Do'),'y':day6}, 
						{'x':moment().subtract(4, 'days').format('ddd MMMM Do'),'y':day5}, 
						{'x':moment().subtract(3, 'days').format('ddd MMMM Do'),'y':day4}, 
						{'x':moment().subtract(2, 'days').format('ddd MMMM Do'),'y':day3}, 
						{'x':moment().subtract(1, 'days').format('ddd MMMM Do'),'y':day2}, 
						{'x':moment().format('ddd MMMM Do'),'y':day1});

**Comment by codesuki** [comment]
I think I know where your problem is. Could you test something?
In your case the X scale will be an d3.ordinalScale which doesn't have an invert() function.
I will try to find a way for this case in the future but if you would change your `x` to be a `Date` object a time scale should be used and invert will work. 

So could you try to add `.toDate()` in all the `'x'` assignments like this:
```
{'x':moment().subtract(6, 'days').format('ddd MMMM Do').toDate(), 'y':day7},
```

**Comment by ryanhca** [comment]
Yes. that worked. I'd like to be able to format the date at some point in the future. but for now for this iteration I'll be able to use the chart with that date format. Let me know if you ever come across a fix. ryanhca@gmail.com is my email. Feel free to reach out. I'd love to look into your library and see if I can't find a fix sometime when I'm not so busy at work. I'll fork it this week and start poking around.

**Comment by ryanhca** [comment]
Also, is there a way to hide labels if there is no data to display? example, I have an active vs inactive piechart and often times there are no "inactive" data types to display, but the label still shows up for the pie chart

**Comment by codesuki** [comment]
Thank you for the praise, I welcome any help improving this library. Just send a pullrequest once you have something going.

About your first question with the date format, I think it should be possible by setting the tick format on the axis. I will have a look at it.

Having an option to remove the label if necessary sounds good. Maybe there is even a possibility to split out the label code from the graph code. Not sure.

What exactly do you mean by having several charts in the same 'parent' ? 
Do you want to show 3 different pie charts at the same time (3 different rings?) 
Do you just want to show different pie chart at the same place?

**Comment by codesuki** [comment]
@ryanhca about your question regarding the timestamp formatting:
I am already doing something similar in the example html file.

```javascript
<LineChart
xAxis={{tickValues: this.state.xScale.ticks(d3.time.day, 2), tickFormat: d3.time.format("%m/%d")}}
```

**Comment by ryanhca** [comment]
The d3.colorscale option is something that I completely looked over! That worked marvelously. The charts are all sibling components. They receive props from the parent's state regarding the data they're displaying. The problem was that the colorScale was adding a new color to each new arc that d3 created. And I'll give the tickFormatting a shot! Thanks again! Fun stuff here!

**Comment by guptasaumya** [comment]
`<div>
                    <BarChart
                        groupedBars
                        data={this.state.aggregatedObject}
                        width={width}
                        height={height}
                        tooltipHtml={tooltip}
                        tooltipMode={'mouse'}
                        colorScale={d3.scaleOrdinal()
                            .domain(['Hits', 'Success', 'Error', 'Timeout'])
                            .range(['#A07EEF', '#78C948', '#2791FF', '#FE0000'])}
                        margin={{ top: 50, bottom: 50, left: 50, right: 50 }}
                    />
                </div>
`
This tooltipHtml feature isn't working for me. I have also declared this variable:

`var tooltip = function(x, y0, y, total) {
 console.log(y.toString())
    return y.toString();
    };
`

the console.log(y.toString()) works here but doesn't show on the UI?

**Comment by theSuiGenerisAakash** [comment]
How is it supposed to work when I am using strings as elements of the array for x-axis ? I am getting the same error @codesuki

**Comment by sanuradhag** [comment]
This issue is still not fixed @codesuki
